ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -1549,8 +1549,8 @@ let migrate_send' ~__context ~vm ~dest ~live:_ ~vdi_map ~vif_map ~vgpu_map
         try
           let _, state = XenopsAPI.VM.stat dbg vm_uuid in
           if Xenops_interface.(state.Vm.power_state = Suspended) then (
-            debug "xenops: %s: shutting down suspended VM" vm_uuid ;
-            Xapi_xenops.shutdown ~__context ~self:vm None
+            debug "xenops: %s: Resuming suspended VM on source host" vm_uuid ;
+            Xapi_xenops.fast_resume ~__context ~self:vm
```

Comment:
This will need to distinguish whether we've already force reset the lifecycle state above (and that is what failed, in which case we can't resume or somthing earlier).
So probably need to move this `with` right before the 'force_state_reset` `if` condition above. And have another `try/with` that does the force shutdown as before.

We might also need to introduce a new synchronization point where the other end can tell us whether it failed to resume the VM there, so that it can be resumed on the source, and we need to wait for that before attempting to destroy the VM on the source.